app/pages/stacking/delegated-stacking/components/duration-cycles-form.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, useState, useMemo } from 'react';
+import { Flex, InputProps, Text, Box } from '@blockstack/ui';
+import { useSelector } from 'react-redux';
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import updateLocale from 'dayjs/plugin/updateLocale';
+import advancedFormat from 'dayjs/plugin/advancedFormat';
+
+import { MAX_STACKING_CYCLES, MIN_STACKING_CYCLES } from '@constants/index';
+import { selectNextCycleInfo } from '@store/stacking';
+import { RootState } from '@store/index';
+import { CircleButton } from '@components/circle-button';
+
+dayjs.extend(duration);
+dayjs.extend(updateLocale);
+dayjs.extend(relativeTime);
+dayjs.extend(advancedFormat);
+
+interface DurationCycleFromProps extends Omit<InputProps, 'form'> {
+  duration: number | null;
+  onUpdate(duration: number): void;
+}
+
+const createCycleArray = () => new Array(12).fill(null).map((_, i) => i + 1);
+
+export const DurationCyclesForm: FC<DurationCycleFromProps> = props => {
+  const { duration, onUpdate } = props;
+
+  const [localCycles, setLocalCycles] = useState(duration ?? 1);
+
+  const { cycleInfo } = useSelector((state: RootState) => ({
+    cycleInfo: selectNextCycleInfo(state),
+  }));
+
+  const cycleLabels = useMemo(() => {
+    if (!cycleInfo) return [];
+    return createCycleArray().map(cycles => {
+      return `
+        ${cycles.toString()} cycle${cycles !== 1 ? 's' : ''}
+        (expires around ${dayjs()
+          .add(cycleInfo.estimateCycleDurationSeconds * cycles, 'seconds')
+          .format('Do MMMM')})`;
+    });
+  }, [cycleInfo]);
+
+  return (
+    <Flex
+      alignItems="center"
+      justifyContent="space-between"
+      mt="base"
+      padding="8px"
+      boxShadow="low"
+      border="1px solid #F0F0F5"
+      borderRadius="8px"
+      onClick={e => {
+        e.stopPropagation();
+        e.preventDefault();
+      }}
+    >
+      <Text alignItems="center" ml="tight" color="ink.900">
+        {cycleLabels[(duration ?? 1) - 1]}
+      </Text>
+      <Box>
+        <CircleButton
+          onClick={e => {
+            e.stopPropagation();
+            setLocalCycles(Math.max(MIN_STACKING_CYCLES, localCycles - 1));
+            onUpdate(Math.max(MIN_STACKING_CYCLES, localCycles - 1));
+          }}
+        >
+          -
+        </CircleButton>
+        <CircleButton
+          ml={[null, 'extra-tight']}
```

Comment:
```suggestion
          ml={['unset', 'extra-tight']}
```

Reply:
Why is this better? Doesn't `null` just not set any value? I'd rather than setting it to undo any existing value.

Reply:
I suppose null can work fine, I typically use `unset` because it's a css value, and haven't ever used the `null` value in the responsive array. I'll resolve this as it's more of just personal preference